nvflare/fuel/f3/streaming/byte_streamer.py

```diff
@@ -190,21 +185,24 @@ def _stop_task(self, task: TxTask, error: StreamError = None, notify=True):
         else:
             # Result is the number of bytes streamed
             task.stream_future.set_result(task.offset)
-        task.stop = True
+
+        self.tx_task_map.pop(task.sid)
```

Comment:
Need to catch the exception before this to ensure this pop to execute (place this in the finally block), otherwise it will result to memory leak again. For example, the self.cell.fire_and_forget() may raise exception.

Reply:
Why do we have two places to touch the map? This could be vulnerable to race conditions.
Consider using a single "task manager" to manage task's lifecycle. You could use a separate thread that monitors the task status. All other places simply update task status. For example, the "stop_task" simply marks the task to be "stopped", and it's up to the "task monitor" to remove it from the map. The task object doesn't have to be removed right away, as long as it is removed. This can avoid the race condition that multiple places try to manipulate the same task object and the map.

Reply:
The sending/receiving sides look very similar. There are 2 different maps: tx_task_map, rx_task_map, each map only have one place that create/remove entries. The earlier implementation uses a separate thread to manage the maps. The only reason for that was some ACK may come after the task has ended. The ACKs are just ignored now so no need for a different thread.